main.cpp

```diff
@@ -904,79 +917,109 @@ int main(int32_t argc, char** argv)
-                                std::printf("Implement it!\r\n");
-                                /*
-                                pcs.width = wndRect.width;
-                                pcs.height = wndRect.height;
-                                fixPCS = true;
-                                */
-                            }
-                            else {
-                                if (!rectIsContained(screenRect, wndRect)) {
-                                    t_timestamp timestamp = PTStoTimestamp(header.pts1);
-                                    std::printf("Window is outside new screen area at timestamp %lu:%02lu:%02lu.%03lu\r\n", timestamp.hh, timestamp.mm, timestamp.ss, timestamp.ms);
-
-                                    uint16_t wndRightPoint    = wndRect.x    + wndRect.width;
-                                    uint16_t screenRightPoint = screenRect.x + screenRect.width;
-                                    if (wndRightPoint > screenRightPoint) {
-                                        corrHor = wndRightPoint - screenRightPoint;
-                                    }
-
-                                    uint16_t wndBottomPoint    = wndRect.y    + wndRect.height;
-                                    uint16_t screenBottomPoint = screenRect.y + screenRect.height;
-                                    if (wndBottomPoint > screenBottomPoint) {
-                                        corrVer = wndBottomPoint - screenBottomPoint;
+                        if (doMove) {
+                            for (int i = 0; i < wds.numberOfWindows; i++) {
+                                t_window *window = &wds.windows[i];
+                                int16_t minDeltaX = -(int16_t)window->WindowsHorPos;
+                                int16_t minDeltaY = -(int16_t)window->WindowsVerPos;
+                                int16_t maxDeltaX = pcs.width - (window->WindowsHorPos + window->WindowsWidth);
+                                int16_t maxDeltaY = pcs.height - (window->WindowsVerPos + window->WindowsHeight);
+                                int16_t clampedDeltaX = std::min(std::max(cmd.move.deltaX, minDeltaX), maxDeltaX);
+                                int16_t clampedDeltaY = std::min(std::max(cmd.move.deltaY, minDeltaY), maxDeltaY);
+
+                                window->WindowsHorPos += clampedDeltaX;
+                                window->WindowsVerPos += clampedDeltaY;
+
+                                for (int j = 0; j < pcs.numCompositionObject; j++) {
+                                    t_compositionObject *object = &pcs.compositionObject[j];
+                                    if (object->windowID != window->windowID) continue;
+
+                                    if (object->objectCroppedFlag == 0x40) {
+                                        object->objCropHorPos += clampedDeltaX;
+                                        object->objCropVerPos += clampedDeltaY;
```

Comment:
I've never seen objCropHorPos and objCropVerPos used in the wild, always being filled with a fixed value, I think you should check if those field are at a default and only change them otherwise, personally I'd also output a warning when changing those

Reply:
I thought that the `objectCroppedFlag` being set to enabled (`0x40`) indicated that these aren't set to default values (`0`?) and should be respected.
In that case, they are some absolute coordinates which should be translated like the subpic itself to keep their relative positions and thus leave the crop unchanged.

Reply:
In my experience that flag is set when the subpic is forced and if you check in ReadPCS and WritePCS the handling of those fields is commented out, i can share you this file which was sent to me to debug this flag behavior, if I remember correctly I never found those field populated, but I may be wrong
[testcropped.zip](https://github.com/MonoS/SupMover/files/14100594/testcropped.zip)

Reply:
Oh, I missed that these fields aren't even read/written currently. Then it's better to leave this to a different PR, perhaps.
I will print a warning instead.